pkg/operator/hive/hive_controller.go

```diff
@@ -254,11 +245,28 @@ func (r *ReconcileHiveConfig) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, nil
 	}
 
-	recorder := events.NewRecorder(r.kubeClient.CoreV1().Events(constants.HiveNamespace), "hive-operator", &corev1.ObjectReference{
+	recorder := events.NewRecorder(r.kubeClient.CoreV1().Events(r.hiveOperatorNamespace), "hive-operator", &corev1.ObjectReference{
 		Name:      request.Name,
-		Namespace: constants.HiveNamespace,
+		Namespace: r.hiveOperatorNamespace,
 	})
 
+	// Ensure the target namespace for hive components exists and create if not:
+	hiveNSName := getHiveNamespace(instance)
+	hiveNamespace := &corev1.Namespace{}
+	if err := r.Client.Get(context.Background(), types.NamespacedName{Name: hiveNSName}, hiveNamespace); err != nil {
```

Comment:
Since you do not need to use the `corev1.Namespace` object, don't bother doing a Get first. Create the namespace and handle the error condition when the namespace already exists.